R/tokens.R

```diff
@@ -264,42 +264,44 @@ tokens.corpus <- function(x,
                            fastestword = tokenize_fastestword)
     
     
-    if (what == "word") {
-        x <- preserve_special(x, split_hyphens = split_hyphens, 
-                               split_tags = FALSE, verbose = verbose)
-        special <- attr(x, "special")
-    }
+    if (!remove_separators && !what %in% c("word", "word1", "character"))
+        warning("remove_separators is always TRUE for this type")
     
     # split x into smaller blocks to reducre peak memory consumption
-    x <- split(x, ceiling(seq_along(x) / 10000))
-    for (i in seq_along(x)) {
-        if (verbose) catm(" ...tokenizing", i, "of", length(x), "blocks\n")
-        temp <- tokenizer_fn(x[[i]], split_hyphens = split_hyphens, verbose = verbose)
-        if (verbose) catm(" ...serializing tokens, ")
-        if (i == 1) {
-            x[[i]] <- serialize_tokens(temp)
-        } else {
-            x[[i]] <- serialize_tokens(temp, attr(x[[i - 1]], "types"))
+    x <- texts(x)
+    x <- split(x, factor(ceiling(seq_along(x) / 10000)))
```

Comment:
Doesn't have to be done here, but at some point the fixed 10000 might best to move to an option.

Reply:
Added a quanteda option `block_size`

Reply:
👍  but how about we call it `tokens_block_size` - it's only for the tokens construction, right?